Frends.Kafka.Produce/Frends.Kafka.Produce/Produce.cs

```diff
@@ -1,212 +1,411 @@
+    {
+        using var producer = new ProducerBuilder<string, string>(producerConfig).Build();
+
+        TopicPartition topicPartition = null;
+        if (partition >= 0)
+            topicPartition = new TopicPartition(topic, new Partition(partition));
+
+        Message<string, string> message = new()
+        {
+            Key = msgKey ?? null,
+            Value = msg
+        };
+
+        if (topicPartition != null)
+            return new Result(true, await producer.ProduceAsync(topicPartition, message, cancellationToken));
+        else
+            return new Result(true, await producer.ProduceAsync(topic, message, cancellationToken));
+    }
+
+    private static async Task<Result> ProduceAvro(string topic, string msgKey, int partition, SchemaRegistry schemaRegistry, ProducerConfig producerConfig, CancellationToken cancellationToken)
+    {
+        var schemaRegistryConfig = new SchemaRegistryConfig()
+        {
+            Url = AssignIfNotNullOrEmpty(schemaRegistry.SchemaRegistryUrl, string.Empty),
+            BasicAuthCredentialsSource = GetBasicAuthCredentialsSource(schemaRegistry.BasicAuthCredentialsSource),
+            EnableSslCertificateVerification = schemaRegistry.EnableSslCertificateVerification,
+            BasicAuthUserInfo = AssignIfNotNullOrEmpty(schemaRegistry.BasicAuthUserInfo, string.Empty),
+            RequestTimeoutMs = schemaRegistry.RequestTimeoutMs,
+            MaxCachedSchemas = schemaRegistry.MaxCachedSchemas,
+        };
+
+        if (!string.IsNullOrEmpty(schemaRegistry.SslCaLocation))
+        {
+            schemaRegistryConfig.SslCaLocation = AssignIfNotNullOrEmpty(schemaRegistry.SslCaLocation, string.Empty);
+            schemaRegistryConfig.SslKeystorePassword = AssignIfNotNullOrEmpty(schemaRegistry.SslKeystorePassword, string.Empty);
+            schemaRegistryConfig.SslKeystoreLocation = AssignIfNotNullOrEmpty(schemaRegistry.SslKeystoreLocation, string.Empty);
+        }
 
-            Message<string, string> message = new()
+        TopicPartition topicPartition = null;
+        if (partition >= 0)
+            topicPartition = new TopicPartition(topic, new Partition(partition));
+
+        using var cachedSchemaRegistryClient = new CachedSchemaRegistryClient(schemaRegistryConfig);
+        using var producer = new ProducerBuilder<string, GenericRecord>(producerConfig)
+            .SetValueSerializer(new AvroSerializer<GenericRecord>(cachedSchemaRegistryClient))
+            .Build();
+
+        var avroSchema = (RecordSchema)Avro.Schema.Parse(schemaRegistry.SchemaJson);
+        JObject recordsJObject = JObject.Parse(schemaRegistry.Records);
+
+        var records = new GenericRecord(avroSchema);
+        AddFields(avroSchema, recordsJObject, records);
+
+        if (topicPartition != null)
+            return new Result(true, await producer.ProduceAsync(topicPartition, new Message<string, GenericRecord> { Key = msgKey, Value = records }, cancellationToken));

```

Comment:
lets actually expand this into several lines. like 
`var message = new Message...`
`var produceResult = await ProduceAsuc...`
`return new Result()`

Otherwise it is really hard to read and understand what is happening.